RenderStateNotation/src/RenderStateNotationParserImpl.cpp

```diff
@@ -408,46 +408,47 @@ RenderStateNotationParserImpl::RenderStateNotationParserImpl(IReferenceCounters*
                     case PIPELINE_TYPE_GRAPHICS:
                     case PIPELINE_TYPE_MESH:
                     {
-                        GraphicsPipelineNotation ResourceDesc{};
-                        AssignPipelineState(ResourceDesc, DefaultPipeline, PipelineType);
-                        Deserialize(Pipeline, ResourceDesc, *m_pAllocator, Callbacks);
-                        VERIFY_EXPR(ResourceDesc.PSODesc.Name != nullptr);
 
-                        m_GraphicsPipelineNames.emplace(HashMapStringKey{ResourceDesc.PSODesc.Name, false}, StaticCast<Uint32>(m_GraphicsPipelineStates.size()));
-                        m_GraphicsPipelineStates.push_back(ResourceDesc);
+                        auto* pResourceDesc = m_pAllocator->Construct<GraphicsPipelineNotation>();
```

Comment:
Prefer using references if the pointer can't be null:
auto& ResourceDesc = *m_pAllocator->Construct<ComputePipelineNotation>();